docs/language.md

```diff
@@ -0,0 +1,18 @@
+# Language Support in Talpa
+
+> This document is incomplete
+
+Talpa was designed with a unique capability to compile for multiple platforms from a single codebase. 
+To do this, Talpa generates a tree which can be parsed into a language with a language compile structure. 
+These structures implement functions that parse objects from the tree into the source code for that language. 
+
+To code language support for Talpa, you will need to know a few things.
+
+- Rust (The language Talpa is written in)
+- Talpa's parser tree layout
+- The language you wish to parse into
+
+To implement the language support, you will need to complete 2 steps.
+
+1. Create and implement the structure in languages/*mylanguage*.rs
```

Comment:
Maybe it's better todo: `languages/*mylanguage*/mod.rs`  
Once i can also add support for a language i want to see if we can compile talpa into rust.  
But doing so is a lot more complicated than js and requires a lot more files than only 1 rust file.